### PR TITLE
[fix #183] fix memory sort

### LINKED ISSUE #183: TiKV-CDC sink data with wrong sequence.
### 1. Describe the bug
 TiKV-CDC sink data with wrong sequence.
### 2. Minimal reproduce step (Required)
1. Deploy two tikv cluster with apiv2, tikv-cdc is included in upstream cluster.
2. Create change feed to sync data from upstream to downstream.
3. Run `ycsb run` with workloada.
4. Wait 20 seconds and kill ycsb, wait cdc sink finish and run checksum between two tikv clusters.

### 3. What did you see instead (Required)

Checksum is different, key cnt and bytes are the same, but crc64xor is different.

### 4. What did you expect to see? (Required)

Checksum should not exactly same.

### 5. What is your migration tool and TiKV version? (Required)

- TiKV CDC:

### PR DESCRIPTION
**Pull request by zeminzhou**
Signed-off-by: zeminzhou <zhouzemin@pingcap.com>

### What problem does this PR solve?

fix memory sort

Issue Number: close #183

Problem Description: **TBD**

### What is changed and how does it work?

memory sort is not stable, replace sort.Slice with sort.StableSlice

### Code changes

- Has exported function/method change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test

### CODE REVIEW

**Review by pingyu — COMMENTED** [review]
Suggest to add test case for this change. Rest LGTM.

**Review by haojinming — APPROVED** [review]

File: cdc/cdc/sorter/unified/sorter_test.go [hunk 1]
```diff
@@ -210,6 +232,14 @@ func testSorter(ctx context.Context, c *check.C, sorter sorter.EventSorter, coun
 					if event.CRTs < lastTs {
 						panic("regressed")
 					}
+					fmt.Println(string(event.RawKV.Key))
```
Comment (pingyu) [inline cdc/cdc/sorter/unified/sorter_test.go:235]: Remove the debug code.

File: cdc/cdc/sorter/unified/sorter_test.go [hunk 2]
```diff
@@ -210,6 +232,14 @@ func testSorter(ctx context.Context, c *check.C, sorter sorter.EventSorter, coun
...
+					if lastEvent != nil &&
+						bytes.Equal(lastEvent.RawKV.Key, event.RawKV.Key) {
+						fmt.Println(string(lastEvent.RawKV.Key))
```
Comment (pingyu) [inline cdc/cdc/sorter/unified/sorter_test.go:238]: Ditto.

**Review by pingyu — COMMENTED** [review]
Rest LGTM.

File: cdc/cdc/model/mounter.go [hunk 3]
```diff
@@ -54,6 +54,25 @@ func (e *PolymorphicEvent) RegionID() uint64 {
...
+
+// ComparePolymorphicEvents compares two events by CRTs, Resolved order.
+// It returns true if and only if i should precede j.
+func ComparePolymorphicEvents(i, j *PolymorphicEvent) bool {
+	if i.CRTs == j.CRTs {
+		if i.IsResolved() {
+			return false
+		} else if j.IsResolved() {
+			return true
+		}
```
Comment (haojinming) [inline cdc/cdc/model/mounter.go:71]: if j.IsResolved() {
	return true
}

**Review by haojinming — APPROVED** [review]

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**